src/filters/simple-filter-directive.js

```diff
@@ -0,0 +1,277 @@
+              title:  'Address',
+              placeholder: 'Filter by Address',
+              filterType: 'text'
+            },
+            {
+              id: 'birthMonth',
+              title:  'Birth Month',
+              placeholder: 'Filter by Birth Month',
+              filterType: 'select',
+              filterValues: ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December']
+            }
+          ],
+          resultsCount: $scope.items.length,
+          appliedFilters: [],
+          onFilterChange: filterChange
+        };
+      }
+    ]);
+  </file>
+</example>
+ */
+angular.module('patternfly.filters').directive('pfSimpleFilter',
+  function ($document) {
+    'use strict';
+    return {
+      restrict: 'A',
+      scope: {
+        config: '='
+      },
+      transclude: false,
+      templateUrl: 'filters/simple-filter.html',
+      controller: function ($scope) {
+        var defaultConfig = {
+          fields: [],
+          resultsCount: 0
+        };
+
+        $scope.setupConfig = function () {
+          $scope.config = $.extend(true, angular.copy(defaultConfig), $scope.config);
+
+          if (!$scope.currentField) {
+            $scope.currentField = $scope.config.fields[0];
+            $scope.config.currentValue = null;
+          }
+
+          if ($scope.config.currentValue === undefined) {
+            $scope.config.currentValue = null;
+          }
+
+          if (!$scope.config.appliedFilters) {
+            $scope.config.appliedFilters = [];
+          }
+        };
+      },
+      link: function (scope, element, attrs) {
+        scope.$watch('config', function () {
+          scope.setupConfig();
+        }, true);
+
+        scope.selectField = function (item) {
+          scope.currentField = item;
+          scope.config.currentValue = null;
+        };
+
+        scope.selectValue = function (filterValue) {
+          scope.addFilter(scope.currentField, filterValue);
+          scope.config.currentValue = null;
+        };
+
+        scope.filterExists = function (filter) {
+          var found = false;
+          scope.config.appliedFilters.forEach(function (nextFilter) {
+            if (nextFilter.title === filter.title && nextFilter.value === filter.value) {
+              found = true;
+              return found;
+            }
+          });
+          return found;
+        };
+
+        scope.addFilter = function (field, value) {
+          var newFilter = {
+            id: field.id,
+            title: field.title,
+            value: value
+          };
+          if (!scope.filterExists(newFilter)) {
+            scope.config.appliedFilters.push(newFilter);
+
+            if (scope.config.onFilterChange) {
+              scope.config.onFilterChange(scope.config.appliedFilters);
+            }
+          }
+        };
+
+        scope.onValueKeyPress = function (keyEvent) {
+          if (keyEvent.which === 13) {
```

Comment:
Also, we don't want to filter as the user types instead of making them hit enter in the first place?  You can easily accomplish this with an angularjs filter.